### PR TITLE
Make namespace configurable for anywhere resources

### PR DESCRIPTION
**Pull request by taneyland**
*Issue #, if available:*

*Description of changes:*
This PR creates a namespace field in the clusterconfig for all "anywhere.eks.amazonaws.com" resources. This namespace is configurable, so the user can specify the name of the namespace to store these resources. If no namespace is specified, it will behave as it does currently & use the "default" namespace.

These change were tested with the following cases (all cases had external etcd configured):
- Create/Delete with Flux enabled
- Upgrade with Flux enabled
- Create/Delete without Flux
- Upgrade without Flux

- EKS-A controller still behaves as expected
- Not specifying namespace still defaults to the "default" namespace

- By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by danbudris** [comment]
I might be missing it in all these files, but we should make sure that we prevent `flux` from modifying the `namespace` attribute of the cluster.

We've got [the validation in the upgrade preflight check of immutable fields](https://github.com/aws/eks-anywhere/pull/177/files#diff-eb164fd49ebb5b09b6e866a9e9bca69e59861aead0fc8293d5c5d7b7607d6503R13), but we'll want to add it to [the immutable fields in the `cluster` webhook](https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/cluster_webhook.go#L71), as well

**Comment by vivek-koppuru** [comment]
> I might be missing it in all these files, but we should make sure that we prevent `flux` from modifying the `namespace` attribute of the cluster.
> 
> We've got [the validation in the upgrade preflight check of immutable fields](https://github.com/aws/eks-anywhere/pull/177/files#diff-eb164fd49ebb5b09b6e866a9e9bca69e59861aead0fc8293d5c5d7b7607d6503R13), but we'll want to add it to [the immutable fields in the `cluster` webhook](https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/cluster_webhook.go#L71), as well

Since we are retrieving objects based on namespace, would change the namespace count as creating a new object instead, which would need to be blocked for the create rather than making that specific field immutable?

File: pkg/api/v1alpha1/object_meta.go [hunk 1]
```diff
@@ -6,5 +6,6 @@ package v1alpha1
 // and https://github.com/kubernetes-sigs/cluster-api/blob/bf790fc2a53614ff5d3405c83c0de0dd3303bb1f/api/v1alpha2/common_types.go#L67-L128
 // as needed.
 type ObjectMeta struct {
-	Name string `json:"name,omitempty"`
+	Name      string `json:"name,omitempty"`
+	Namespace string `json:"namespace,omitempty"`
```
Comment (vivek-koppuru) [inline pkg/api/v1alpha1/object_meta.go:10]: We don't need to output the namespace as part of the generate command, so this isn't required right?

File: pkg/executables/kubectl.go [hunk 2]
```diff
@@ -578,7 +583,7 @@ func (k *Kubectl) RemoveAnnotationInNamespace(ctx context.Context, resourceType,
 }
 
 func (k *Kubectl) GetEksaCluster(ctx context.Context, cluster *types.Cluster) (*v1alpha1.Cluster, error) {
-	params := []string{"get", eksaClustersResourceType, cluster.Name, "-o", "json", "--kubeconfig", cluster.KubeconfigFile}
+	params := []string{"get", "clusters", "-A", "-o", "jsonpath={.items[0]}", "--kubeconfig", cluster.KubeconfigFile, "--field-selector=metadata.name=" + cluster.Name}
```
Comment (vivek-koppuru) [inline pkg/executables/kubectl.go:586]: If you do this, you can ensure that it will always pick the right cluster and then compare namespace values from that? What happens if there are multiple clusters with the same name in different namespaces (which is an edge case)? We may want a check for that to error out?

**Review by feyyyyylo — APPROVED** [review]

**Review by vivek-koppuru — APPROVED** [review]
Looks good to me overall, but let's add a e2e test for this that specifies a different namespace from default and tests flux and upgrade.

File: pkg/validations/upgradevalidations/immutableFields.go [hunk 3]
```diff
@@ -20,6 +20,12 @@ func ValidateImmutableFields(ctx context.Context, k ValidationsKubectlClient, cl
 		return fmt.Errorf("cluster name is immutable")
 	}
 
+	if prevSpec.Namespace != spec.Namespace {
+		if !(prevSpec.Namespace == "default" && spec.Namespace == "") {
```
Comment (vivek-koppuru) [inline pkg/validations/upgradevalidations/immutableFields.go:24]: Do we need to have this same check for GitOps and OIDC? Or is this not the case because it would fail in validations earlier anyways? If that's the case, we technically don't need the checks above for VSphere if this check here covers it for all the config anyways.

**Review by vivek-koppuru — APPROVED** [review]

File: pkg/validations/upgradevalidations/immutableFields.go (hunk 3 above)
Comment (taneyland) [reply pkg/validations/upgradevalidations/immutableFields.go:24]: That's true. These cases are covered in earlier validations. I removed the unnecessary checks

**Review by vivek-koppuru — APPROVED** [review]
/lgtm
